scripts/west_commands/run_common.py

```diff
@@ -201,6 +322,84 @@ def do_run_common_image(command, user_args, user_runner_args, build_dir=None):
     # parsing, it will show up here, and needs to be filtered out.
     runner_args = [arg for arg in user_runner_args if arg != '--']
 
+    # Check if there are any commands that should only be ran once per board
+    # and if so, remove them for all but the first iteration of the flash
+    # runner per unique board name.
+    if len(used_cmds) > 0 and len(runner_args) > 0:
+        l = len(runner_args) - 1
+        while l >= 0:
+            i = 0
+            while i < len(used_cmds):
+                if used_cmds[i].command == runner_args[l] and (runner_name in used_cmds[i].runners \
+                                                               or 'all' in used_cmds[i].runners):
+                    # Check if board is here
+                    found = False
+                    for check in used_cmds[i].boards:
+                        if board == check:
+                            found = True
+                            break
+
+                    if found is True:
+                        # Check if this is a first or last run
+                        if used_cmds[i].first is False:
+                            # For last run instances, we need to check that this really is the last
+                            # image of all boards being flashed
+                            for board in used_cmds[i].boards:
+                                if board_image_count[board].flashed != board_image_count[board].total:
+                                    found = False
+                                    break
```

Comment:
This overwrites `board` so it's no longer equal to `cache['CACHED_BOARD']` in the function's scope.
It's safer to use a generator instead, for example:
```suggestion
                            found = all(
                                board_image_count[board].flashed == board_image_count[board].total
                                for board in used_cmds[i].boards
                            )
```

Reply:
Has been reworked